src/Plotting.jl

```diff
@@ -0,0 +1,173 @@
+"""
+    Plotting
+
+Module providing plotting functionality for ClimaCoupler simulations.
+
+The actual implementations are provided by the `ClimaCouplerMakieExt` extension
+when Makie.jl and related packages are loaded.
+"""
+module Plotting
+
+import ..Interfacer
+import ..ConservationChecker
+
+export make_diagnostics_plots,
+    make_ocean_diagnostics_plots,
+    debug,
+    plot_global_conservation,
+    compute_leaderboard,
+    compute_pfull_leaderboard,
+    scatter_plot,
+    param_versus_iter_plot,
+    debug_plot_fields
+
+"""
+    make_diagnostics_plots(
+        output_path::AbstractString,
+        plot_path::AbstractString;
+        output_prefix = "",
+    )
+
+Stub method for plotting diagnostic variables.
+See the ClimaCouplerMakieExt extension for the actual implementation.
+"""
+function make_diagnostics_plots(
```

Comment:
make stubs simpler and add _init_ with extension_fns like [this](https://github.com/CliMA/ClimaUtilities.jl/blob/2046cafc03354628a43f525bafffe29f69e0a006/src/Regridders.jl#L75)